test/unit/org/apache/cassandra/tools/ToolRunner.java

```diff
@@ -273,228 +109,466 @@ public void checkPermission(Permission perm, Object context)
         }
     }
 
-    public boolean isRunning()
+    private static final class StreamGobbler<T extends OutputStream> implements Runnable
```

Comment:
is there a reason to make this generic?  I don't see us doing anything with it; I am ok if you want to keep it there.

Reply:
did I do this in my POC?  If so I think I did it because I had some callback logic I eventually removed as it turned out to be less useful.

Reply:
Yeah you're right. But it doesn't hurt and it's going to be there already in case we ever need it :shrug: